mautic/mautic
# Integration bundle improvements

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 



#### Description:

Currently there is no validation for data type and/or length for fields when the "integration to Mautic" sync process runs. The lack of such validation may result in crashing of the synchronization. The user cannot see any of these errors. This PR tackles this issue by adding the data validation. In case there are any validation errors the given fields are not synced and a user is notified about the errors via UI notifications.

This PR also adds the following improvements:
1. Adds new integrations events: 
    - `'mautic.integration.INTEGRATION_BATCH_SYNC_COMPLETED_INTEGRATION_TO_MAUTIC'`
    - `'mautic.integration.INTEGRATION_BATCH_SYNC_COMPLETED_MAUTIC_TO_INTEGRATION'`
2. Multiselect custom fields DB type has been changed to `TEXT (65535)`

#### Steps to test this PR:



1. As there is no existing integration there is no way to test the changes in the integration bundle.
2. Make sure UI notification are working as expected.
3. Make sure Multiselect custom fields work as expected.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11303"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
